### PR TITLE
feat: support dplayer

### PR DESCRIPTION
**Pull request by SteveYuOWO**
### <samp>🤖 Generated by Copilot at 63920a4</samp>

This pull request adds support for uploading and displaying video files in the blog editor and posts. It introduces three new dependencies, `file-size`, `rc-dplayer`, and `@types/file-size`, and two new files, `src/components/ui/DPlayer.tsx` and `src/markdown/rehype-video.ts`. It also modifies the `package.json`, `pnpm-lock.yaml`, `src/markdown/index.ts`, `src/pages/dashboard/[subdomain]/editor.tsx`, and `src/lib/constants.ts` files to enable video file handling and rendering.

### <samp>🤖 Generated by Copilot at 63920a4</samp>

> _We're sailing on the IPFS sea, with videos in our blog_
> _We use the `DPlayer` and the `rehypeVideo` cog_
> _We check the file size with the `filesize` tool_
> _And hoist the `MAXIMUM_FILE_SIZE` as our golden rule_

### WHY
Support dplayer

### <samp>🤖 Generated by Copilot at 63920a4</samp>

*  Add `file-size` dependency and types to enable file size formatting and validation in the editor page ([link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R72), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R154), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR130-R132), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR372-R374), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR4601-R4604), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR8219-R8222), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-8c943ccfb5928fff63708e99eaafa93f8845c9405ed8e4b32afed23e5003025aR11-R12), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-842ea70ef7b4f5946fc6938a7a1eba15216a05a8bd9de0d2963aec6b8e0753c0R2), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-842ea70ef7b4f5946fc6938a7a1eba15216a05a8bd9de0d2963aec6b8e0753c0R51), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-842ea70ef7b4f5946fc6938a7a1eba15216a05a8bd9de0d2963aec6b8e0753c0L353-R372))
* Add `rc-dplayer` dependency and component to enable video playback in the markdown renderer ([link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R98), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR208-R210), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR5862-R5871), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR5987-R5990), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR7337-R7346), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR8289-R8295), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR8867-R8870), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR11519-R11522), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR11622-R11638), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR13918-R13921), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-7c1fab6e76704099e2a3a0ab559952be70c731d9db4c59e8039444a4f282891bR1-R36), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-733efe578c40ff2e6b0770350d988ec422a6d3ad9f1543b2bcda58e0ed546fdcR33), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-733efe578c40ff2e6b0770350d988ec422a6d3ad9f1543b2bcda58e0ed546fdcR229))
* Add `rehypeVideo` plugin to convert IPFS URLs to gateway URLs for video elements in the HTML tree ([link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-733efe578c40ff2e6b0770350d988ec422a6d3ad9f1543b2bcda58e0ed546fdcR46), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-733efe578c40ff2e6b0770350d988ec422a6d3ad9f1543b2bcda58e0ed546fdcR146), [link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-fbf90c774b9449a3418eda2d165fec41cf3855cf5320bdb2951ea5c673e87296R1-R36))
* Modify `handleUpload` function to allow video file upload and insertion to the editor content ([link](https://github.com/Crossbell-Box/xLog/pull/497/files?diff=unified&w=0#diff-842ea70ef7b4f5946fc6938a7a1eba15216a05a8bd9de0d2963aec6b8e0753c0R409-R414))

### CLAIM REWARDS

For first-time contributors, please leave your xLog address and Discord ID below to claim your rewards.

### CODE REVIEW

**Comment by SteveYuOWO** [comment]
#362 Dplayer support. I limited the upload size to 100M in my pr.

**Comment by SteveYuOWO** [comment]
image

**Comment by SteveYuOWO** [comment]
confused the lib pf rc-dplayer.
It doesn't seems to support next.js.
Create a issue for [reproduction](https://github.com/tianfeng98/rc-dplayer/issues/1).

**Comment by SteveYuOWO** [comment]
resolved by dynamic import 

> confused the lib of rc-dplayer. It doesn't seems to support next.js. Create a issue for [reproduction](https://github.com/tianfeng98/rc-dplayer/issues/1).

**Comment by DIYgod** [comment]
Thank you so much, but we just completed the migration of the app router, which caused some conflicts. Can you help resolve the conflicts with the dev branch?

**Comment by SteveYuOWO** [comment]
> Thank you so much, but we just completed the migration of the app router, which caused some conflicts. Can you help resolve the conflicts with the dev branch?

Okay. Thanks for your response. The conflict has been resolved.

File: src/app/dashboard/[subdomain]/editor/page.tsx [hunk 1]
```diff
@@ -348,9 +350,22 @@ export default function SubdomainEditor() {
...
           !file.type.startsWith("image/") &&
-          !file.type.startsWith("audio/")
+          !file.type.startsWith("audio/") &&
+          !file.type.startsWith("video/")
         ) {
-          throw new Error("You can only upload images and audios")
+          throw new Error("You can only upload images, audios and videos")
+        }
+
+        const uploadFilesize = filesize(file.size).to("MB")
```
Comment (DIYgod) [inline src/app/dashboard/[subdomain]/editor/page.tsx:359]: What is the function of `filesize` here, why not just use `file.size`?

File: src/app/dashboard/[subdomain]/editor/page.tsx [hunk 2]
```diff
@@ -389,6 +404,12 @@ export default function SubdomainEditor() {
               } ${cover ? `cover="${cover}"` : ""}><audio>\n`,
             ),
           )
+        } else if (file.type.startsWith("video/")) {
+          view?.dispatch(
+            view.state.replaceSelection(
+              `\n<video>\n\t<source src=\"${key}\" type\"${file.type}\" />\n</video>\n`,
```
Comment (DIYgod) [inline src/app/dashboard/[subdomain]/editor/page.tsx:410]: Missing a `=` after `type`

**Review by DIYgod — CHANGES_REQUESTED** [review]

File: src/app/dashboard/[subdomain]/editor/page.tsx (hunk 2 above)
Comment (SteveYuOWO) [reply src/app/dashboard/[subdomain]/editor/page.tsx]: @DIYgod = has been appended

File: src/app/dashboard/[subdomain]/editor/page.tsx (hunk 1 above)
Comment (SteveYuOWO) [reply src/app/dashboard/[subdomain]/editor/page.tsx:359]: https://www.npmjs.com/package/file-size
This package provides storage unit conversion. Convert `bytes` to `mb` here.

Comment (SteveYuOWO) [reply src/app/dashboard/[subdomain]/editor/page.tsx:359]: The`filesize` package is 1.6 kb. Whether bytes needs to be manually evaluated like `file.size / 1024 / 1024`?

**Comment by DIYgod** [comment]
Thanks so much!

**Pull request merged**